### PR TITLE
pyplot.subplots() accepts subplot arguments through **kwargs

### LINKED ISSUE #1460: make subplots smarter with regards to subplot_kw and figure_kw
While working on #1458, it occurred to me that it might not be a bad idea to process the keyword arguments given to `plt.subplots` and pass them along in the appropriate direction. As I alluded to in #1458, currently, to replace lines like:

   fig = plt.figure(figsize=(10,20))
   ax = plt.subplot(111, axisbg='w')

you have to do an awkward thing to subplot's keyword arguments, since, `subplot*s*` passes its keyword arguments to plt.figure.

```
fig, ax = plt.subplots(figsize=(10,20), subplot_kw=dict(axisbg='w'))
```

but, at least as I was going through our examples, all of the subplot keyword arguments were pretty unambiguously named, and could not be interpreted as figure keyword arguments. So there's no reason why we shouldn't make the next line work the same as the two examples above:

```
fig, ax = plt.subplots(figsize=(10,20), axisbg='w')
```

in our docs, plt.axes takes these kwargs:

```
axisbg    color          the axes background color
frameon   [True|False]   display the frame?
sharex    otherax        current axes shares xaxis attribute with otherax
sharey    otherax        current axes shares yaxis attribute with otherax
polar     [True|False]   use a polar axes?
```

and plt.subplot says it deals with `axisbg`, `polar`, and `projection`

### PR DESCRIPTION
**Pull request by sabulikia**
Fixes #1460  

## PR Summary
`pyplot.subplots()` now accepts subplot arguments through **kwargs without the need of using dictionaries. 

```
figure1, axis1 = plt.subplots(figsize=(10,20), polar=True)
figure2, axis2 = plt.subplots(figsize=(10,20), subplot_kw=dict(polar=True)) # this still works
```

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
Wow, that is an old issue!

I share @QuLogic 's concerns about hard-coding a sub-set of kwargs to be hoisted up to `subplot_kw`.  Right now, there is a clear seperation: subplot kwargs go in a dict, fig_kw go as "normal" kwarg so there is no ambiguity over where the input is going to get routed.  By special casing just _some_ of the subplot keyword arguments we are setting our selves up bugs in the future of the form "This kwarg X works, but kwarg Y doesn't".

This also has the logical extension that we should route some keys to `gridspec_kw` if we know they should go there which would make everything more complex.

In short, I am worried that half measures will make things more confusing and inconsistent and the full solution may grow too complex.

**Comment by sabulikia** [comment]
I checked `add_subplot()`'s documentation and it mentions "projection", "polar", "sharex", "sharey", and "label". In `pyplot.subplot()`'s signature  the "sharex" and "sharey" arguments are explicitly passed, so I just extracted the three remaining ones from `**kwargs`. So I'm a little confused about what it is that I'm missing.

**Comment by sabulikia** [comment]
I checked add_subplot()'s documentation and it mentions "projection", "polar", "sharex", "sharey", and "label". In pyplot.subplot()'s signature the "sharex" and "sharey" arguments are explicitly passed, so I just extracted the three remaining ones from **kwargs. 
Just to make sure I understand the issue correctly: The issue with this is that it's not extendable and everytime the arguments need to be changed, the list needs to be updated as well. Also extending this idea to `gridspec_kw` could make things more complicated. Correct? So your suggestion is to look for a more fundamental solution?

**Comment by tacaswell** [comment]
@sabulikia That is a reasonable summary.

 On one hand `add_subplot`, in addition to the keywords it consumes, also pass`**kwargs` through to the subplots class factory.   This means we can never fully get rid of `subplots_kw` (as it is technically possible for the user to provide their on Axes sub-class (which something like windrose [1] does).    On the flip side `plt.figure` also takes `**kwargs` and passes them onto `new_figure_manager` which also take **kwargs and pasess them on.  Thus we can't do the inverse and identify all of the kwargs that should go to `figure` by name (and hence bundle the rest of them into the subplots_kw).   

The fundamental issue is we have two things that consume all of the "extra" keyword arguments and are user-plug able.  I do not think there is a more fundamental solution possible here, but I would be happy to be proved wrong!

It seems simpler to say "input to subplots goes here and input to `Figure` goes there" than to add "...except for XYZ".   Any one exception is easy to remember, but en-masse they are huge sources of friction.

[1] https://github.com/python-windrose/windrose
